src/app/content/utils/canonicalUrl.ts

```diff
@@ -14,29 +15,50 @@ export async function getCanonicalUrlParams(
   pageId: string,
   bookVersion?: string
 ) {
-  const bookDefaultMap = [[book.id, {}]] as Array<[string, ObjectLiteral<undefined>]>;
   const getBook = makeUnifiedBookLoader(archiveLoader, osWebLoader);
-  const bookVersionFromConfig = getBookVersionFromUUIDSync(book.id);
-  const canonicals = ([
-    ...(CANONICAL_MAP[book.id] || []),
-    ...(bookVersionFromConfig && bookVersion === bookVersionFromConfig.defaultVersion ? bookDefaultMap : []),
-    // use the current book as a last resort if it has the same version as in books config
-  ]).filter(([id]) => !!getBookVersionFromUUIDSync(id));
 
-  for (const [id, CANONICAL_PAGES_MAP] of canonicals) {
-    const version = assertDefined(
-      getBookVersionFromUUIDSync(id),
-      `We've already filtered out books that are not in the BOOK configuration`
-    ).defaultVersion;
-    const canonicalBook = book.id === id  && hasOSWebData(book) ? book : await getBook(id, version);
-    const mappedPageId = CANONICAL_PAGES_MAP[pageId] || pageId;
-    const treeSection = findArchiveTreeNodeById(canonicalBook.tree, mappedPageId);
+  const getCanonicalMap = (bookId: string) => {
+    const bookDefaultMap = [[bookId, {}]] as Array<[string, ObjectLiteral<undefined>]>;
+    const bookVersionFromConfig = getBookVersionFromUUIDSync(bookId);
+    return ([
+      ...(CANONICAL_MAP[bookId] || []),
+      ...(bookVersionFromConfig && bookVersion === bookVersionFromConfig.defaultVersion ? bookDefaultMap : []),
```

Comment:
If a non-default bookVersion of a book that is not in the CANONICAL_MAP  is used here, getCanonicalMap will return []. The same can happen if none of the books returned are in BOOKS. This can be a problem for the loop below.